### DISCUSSION TITLE
Using ChatCompletion instead of TextCompletion

### DISCUSSION THREAD

**Discussion by kevmcdonk** [microsoft/teams-sdk#284] [Q&A] [answered]
I’ve been trying to use the GPT4 models but from what I can tell, Teams AI only uses Completions and not Chat currently. Looking at Semantic Kernel, I think it’s possible to use the newer models with Chat using ChatCompletion but can’t see where this is configured. Am I missing something or is this not possible?

The reason for this is that I am finding with completions that it too often returns answers as code instead of natural text that I asked for in the prompt. I am trying to ask the model to analyze some MS Graph JSON response and provide advice on the output but it always comes back as code examples.

**Comment by corinagum** [comment, marked as answer]
Hi @kevmcdonk, your model configuration should be sufficient in indicating what kind of responses are provided. 
[OpenAIClient](https://github.com/microsoft/teams-ai/blob/7d186ec32e26f1544aaa06b70fdd0e592085a9d3/js/packages/teams-ai/src/OpenAIClients/OpenAIClient.ts#L88) | [OpenAIPlanner](https://github.com/microsoft/teams-ai/blob/7d186ec32e26f1544aaa06b70fdd0e592085a9d3/js/packages/teams-ai/src/OpenAIPlanner.ts#L86)

If this is not what you are experiencing, could you share the code snippet of your client setup?

**Reply by kevmcdonk** [reply]
I think I have found the key - apparently the model name needs to start with GPT- to use the chat models. Going to try it tonight

**Reply by corinagum** [reply]
Yes that's correct - I should have linked to the OpenAIPlanner as well. I'm adding that link for future devs with the same question.

**Reply by kevmcdonk** [reply]
Tested and all works! Thanks

**Comment by corinagum** [comment]
I'm going to go ahead and lock this discussion a little early, since @kevmcdonk confirmed the solution. This will remain open for discoverability, but reduce touch points that might fall through the cracks. :)
